File: app/javascript/channels/game_channel.js
import consumer from "channels/consumer"

const gameId = document.getElementById("game-id").value;

// consumer.subscriptions.create({ channel: "GameChannel", game_id: gameId }, {
//   connected() {
//     console.log(`connected to games channel with id ${gameId}`);
//     // Called when the subscription is ready for use on the server
//   },

//   disconnected() {
//     // Called when the subscription has been terminated by the server
//   },

//   received(data) {
//     console.log("received data", data);
//   }
// });

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  def subscribed
    Rails.logger.info("hello games channel")
    Rails.logger.info("params: #{params}")

    game = Game.find(params[:game_id])
    stream_for game
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
